pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/virtualcolumn/VirtualColumnIndexContainer.java

```diff
@@ -46,53 +44,18 @@ public VirtualColumnIndexContainer(ForwardIndexReader<?> forwardIndex, InvertedI
     _dictionary = dictionary;
   }
 
+  @Nullable
   @Override
-  public ForwardIndexReader<?> getForwardIndex() {
-    return _forwardIndex;
-  }
-
-  @Override
-  public InvertedIndexReader<?> getInvertedIndex() {
-    return _invertedIndex;
-  }
-
-  @Override
-  public RangeIndexReader<?> getRangeIndex() {
-    return null;
-  }
-
-  @Override
-  public TextIndexReader getTextIndex() {
-    return null;
-  }
-
-  @Override
-  public TextIndexReader getFSTIndex() {
-    return null;
-  }
-
-  @Override
-  public JsonIndexReader getJsonIndex() {
-    return null;
-  }
-
-  @Override
-  public H3IndexReader getH3Index() {
-    return null;
-  }
-
-  @Override
-  public Dictionary getDictionary() {
-    return _dictionary;
-  }
-
-  @Override
-  public BloomFilterReader getBloomFilter() {
-    return null;
-  }
-
-  @Override
-  public NullValueVectorReader getNullValueVector() {
+  public <I extends IndexReader, T extends IndexType<?, I, ?>> I getIndex(T indexType) {
```

Comment:
Could we do similar to PhysicalColumnIndexCreator and have a map<IndexTyoe, Reader>?